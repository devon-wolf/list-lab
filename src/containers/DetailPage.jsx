import React, { Component } from 'react';
import style from './DetailPage.css';
import VillagerDetail from '../components/displays/VillagerDetail';
import { fetchVillagerById } from '../services/fetchVillagers';

export default class DetailPage extends Component {
	state = {
		loading: true,
		villager: {}
	}

	componentDidMount = async () => {
		const villager = await fetchVillagerById(this.props.match.params.villager);
		this.setState({ loading: false, villager });
	}

	render() {
		const { name, image, phrase } = this.state.villager;

		return (
			<div className={style.detailPage}>
				{this.state.loading
					?	<p>Loading!</p>
					:	<VillagerDetail
							name={name}
							image={image}
							phrase={phrase} 
						/>
				}
			</div>
		)
	}
}
