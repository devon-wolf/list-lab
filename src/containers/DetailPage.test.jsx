import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
// import { MemoryRouter } from 'react-router-dom';
import DetailPage from './DetailPage';

describe('Detail page', () => {
	afterEach(() => cleanup());

	it('renders the detail page', async () => {
		render(
		
				<DetailPage 
					match={{
						params: {
							villager: '5f5fb4bbbfd05c2aed82e46b'
						}
					}}
				/>
		
		);
		
		screen.getByText('Loading!');

		const div = await screen.findByRole('villager-detail');

		expect(div).toBeInTheDocument();
		expect(div).toMatchSnapshot();
	});

})