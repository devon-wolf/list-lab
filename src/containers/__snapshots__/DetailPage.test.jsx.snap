// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail page renders the detail page 1`] = `
<div
  role="villager-detail"
>
  <h3>
    Anicotti
  </h3>
  <img
    alt="Anicotti"
    class="villagerImg"
    src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/80/Anicotti_NH.png/revision/latest?cb=20200802164003"
  />
  <p>
    cannoli
  </p>
</div>
`;
